sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -457,17 +457,33 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
       CompileOpts += std::string(backend_option);
     }
   }
-  if ((PlatformImpl->getBackend() == backend::ext_oneapi_level_zero ||
+  bool IsIntelGPU =
+      (PlatformImpl->getBackend() == backend::ext_oneapi_level_zero ||
        PlatformImpl->getBackend() == backend::opencl) &&
-      std::all_of(Devs.begin(), Devs.end(),
-                  [](const device &Dev) { return Dev.is_gpu(); }) &&
-      Img.getDeviceGlobals().size() != 0) {
+      std::all_of(Devs.begin(), Devs.end(), [](const device &Dev) {
+        return Dev.is_gpu() &&
+               Dev.get_info<info::device::vendor_id>() == 0x8086;
+      });
+  if (IsIntelGPU && Img.getDeviceGlobals().size() != 0) {
     // If the image has device globals we need to add the
     // -ze-take-global-address option to tell IGC to record addresses of these.
     if (!CompileOpts.empty())
       CompileOpts += " ";
     CompileOpts += "-ze-take-global-address";
   }
+  if (!CompileOptsEnv) {
+    static const char *TargetCompileFast = "-ftarget-compile-fast";
+    if (auto Pos = CompileOpts.find(TargetCompileFast);
+        Pos != std::string::npos) {
+      const char *BackendOption = nullptr;
+      PlatformImpl->getBackendOption(TargetCompileFast, &BackendOption);
+      auto OptLen = strlen(TargetCompileFast);
+      if (IsIntelGPU && BackendOption && BackendOption[0] != '\0')
+        CompileOpts.replace(Pos, OptLen, BackendOption);
+      else
+        CompileOpts.erase(Pos, OptLen);
```

Comment:
It looks like your intention was to ignore this option if backend is not an Intel GPU.
But when you call getBackendOption for CUDA/HIP plugins it will result in exception because those plugins don't know about this new option.